pkg/store/proxy_heap.go

```diff
@@ -293,6 +293,19 @@ func (l *lazyRespSet) Empty() bool {
 	l.bufferedResponsesMtx.Lock()
 	defer l.bufferedResponsesMtx.Unlock()
 
+	// NOTE(GiedriusS): need to wait here for at least one
+	// response so that we could build the heap properly.
+	if l.noMoreData && len(l.bufferedResponses) == 0 {
+		return true
+	}
+
+	for len(l.bufferedResponses) == 0 {
```

Comment:
Do we need to unlock the mutex here before waiting?

Reply:
No, Wait() already takes care of everything for us: https://pkg.go.dev/sync#Cond.Wait.